relays/lib-substrate-relay/src/lib.rs

```diff
@@ -91,18 +91,21 @@ impl<AccountId> TaggedAccount<AccountId> {
 }
 
 /// Batch call builder.
-pub trait BatchCallBuilder<Call>: Send {
+pub trait BatchCallBuilder<Call>: Clone + Send {
 	/// Create batch call from given calls vector.
 	fn build_batch_call(&self, _calls: Vec<Call>) -> Call;
 }
 
 /// Batch call builder constructor.
-pub trait BatchCallBuilderConstructor<Call> {
+pub trait BatchCallBuilderConstructor<Call>: Clone {
+	/// Call builder, used by this constructor.
+	type CallBuilder: BatchCallBuilder<Call>;
```

Comment:
An alternative to this would be to add a method to `BatchCallBuilder`: 
`fn clone_box(&self) -> Box<dyn BatchCallBuilder<Call>>;`

And then:
```
impl<Call> Clone for Box<dyn BatchCallBuilder<Call>> {
	fn clone(&self) -> Box<dyn BatchCallBuilder<Call>> {
		self.clone_box()
	}
}
```

Also we would need to use `#[derive(CloneNoBound)]` instead of `#[derive(Clone)]` for `BatchProofTransaction`.

Reply:
Do you think it'd be better? I mean - I'm more in favor of associated types than `Box<dyn>` :) But if you think it'd be better - I could explore and fix it

Reply:
Not sure which version is best. The associated type should be safe also since for the `()` implementation of `BatchCallBuilderConstructor`, `new_builder` will always return `None`. I feel like with the `Box<dyn>` it would be safest, but it's true that there are cons to this as well. 

Anyway, I'm ok with merging it like this (with associated types) and we can revisit the decision later.